src/ray/core_worker/transport/direct_actor_transport.h

```diff
@@ -265,20 +265,50 @@ class FiberEvent {
   bool ready_ = false;
 };
 
+class FiberSemaphore {
+ public:
+  FiberSemaphore(int num) : num_(num) {}
+
+  // Enter the semaphore. Wait fo the value to be > 0 and decrement the value.
+  void Acquire() {
+    std::unique_lock<boost::fibers::mutex> lock(mutex_);
+    cond_.wait(lock, [this]() { return num_ > 0; });
+    num_ -= 1;
+  }
+
+  // Exit the semaphore. Increment the value and notify other waiter.
+  void Release() {
+    {
+      std::unique_lock<boost::fibers::mutex> lock(mutex_);
+      num_ += 1;
+    }
+    // TODO(simon): This not does guarantee to wake up the first queued fiber.
+    // This could be a problem for certain workloads.
```

Comment:
how about adding a sentence on that there's no guarantee on task ordering because of it, that's probably more clear to people reading on this?